core/src/main/scala/it/cwmp/utils/VertxServer.scala

```diff
@@ -0,0 +1,91 @@
+package it.cwmp.utils
+
+import io.vertx.lang.scala.ScalaVerticle
+import io.vertx.scala.core.http.{HttpServerRequest, HttpServerResponse}
+import io.vertx.scala.ext.web.{Router, RoutingContext}
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+/**
+  *
+  * This is a Utility trait that simplifies the creation and management of a vertx server.
+  *
+  * @author Eugenio Pierfederici
+  */
+trait VertxServer extends ScalaVerticle {
+  this: Loggable =>
+
+  protected def serverPort: Int
+
+  protected def initRouter(router: Router): Unit
+
+  protected def initServer: Future[_] = Future.successful(())
+
+  override def startFuture(): Future[_] = {
+    val router = Router.router(vertx)
+    initRouter(router)
+    initServer.flatMap(_ =>
+      vertx.createHttpServer()
+        .requestHandler(router.accept _)
+        .listenFuture(serverPort))
+    .andThen {
+      case Success(_) => logger.info(s"RoomsService listening on port: $serverPort")
+      case Failure(ex) => logger.error(s"Cannot start service on port: $serverPort", ex)
+    }
+  }
+
+  /**
+    * Utility method to send back responses
+    *
+    * @param routingContext the routing context in wich to send the error
+    * @param httpCode       the http code
+    * @param message        the message to send back
+    */
+  protected def sendResponse(httpCode: Int,
+                           message: String = null)
+                          (implicit routingContext: RoutingContext): Unit = Option(message) match {
+    case Some(messageString) =>
+      logger.info(s"Sending $httpCode response to client with message: $messageString")
+      response.setStatusCode(httpCode).end(messageString)
+    case None => response.setStatusCode(httpCode).end()
```

Comment:
Esternamente al matching si può fare response.setStatusCode(httpCode) poi in caso some si fa end(message) in caso none solo end()
Il logging lo metterei da entrambe le parti, sennè sembra che se non c'è un messaggio non venga inviata la risposta